ports/atmel-samd/reset.c

```diff
@@ -32,6 +32,7 @@
 void reset(void) {
     filesystem_flush();
     NVIC_SystemReset();
+    for (;;) {} // unreachable
```

Comment:
Does this and the others do something?

Reply:
Since e.g., NVIC_SystemReset never returns, this creates an unreachable 2 byte opcode that jumps to itself.  However, this allows the compiler to conclude that reset() never returns, as promised.  A more correct change would be to actually let gcc understand that NVIC_SystemReset does not return, but I don't know how to do that.

Reply:
There is a macro `NORETURN` which expands to `__atribute__((noreturn))`, which is described on this page:
https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html#index-noreturn-function-attribute

Reply:
Yes, I should have used the macro instead of the __attribute__ here where I added it: https://github.com/adafruit/circuitpython/blob/dfdfbc4e01dc99c1fa90a9efe5675c02672b3cc5/ports/atmel-samd/reset.h#L39

gcc is complaning because of the NEW noreturn attribute; when it looks at the function definition it thinks it detects the function DOES return, because NVIC_SystemReset is not noreturn.

Reply:
>because NVIC_SystemReset is not noreturn.

That is the kind of thing to fix in the asf4 repo, I'd say.